packages/artifact/src/internal/upload/upload-artifact.ts

```diff
@@ -67,34 +67,38 @@ export async function uploadArtifact(
       'CreateArtifact: response from backend was not ok'
     )
   }
-
+  // Create the zipupload stream for use in blob upload
   const zipUploadStream = await createZipUploadStream(
     zipSpecification,
     options?.compressionLevel
-  )
+  ).catch(err => {
+    throw new InvalidResponseError(
+      `createZipUploadStream: response from backend was not ok: ${err}`
```

Comment:
This error message might be a bit misleading since errors can come from that function without reaching out to a backend.